### PR TITLE
Add node for radar object tracking

### PR DESCRIPTION
**Pull request by PawelLiberadzki**
Introduces RadarTrackObjectsNode, that should be placed right after RadarPostprocessPointsNode. It takes radar detections and groups them into objects. Provides XYZ output that contain object centers. Additionally, public method is provided for this node, that return list of object state structures. With current implementation, object states contain mostly mockup data.

### CODE REVIEW

File: src/math/RuningStats.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Robotec.AI
```
Comment (prybicki) [inline src/math/RuningStats.hpp:1]:
```suggestion
// Copyright 2024 Robotec.AI
```

File: src/graph/RadarTrackObjectsNode.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+	objectStates.clear(); // TODO(Pawel): Remove this when we start working on tracking.
+
+	struct TrackedObject
+	{
+		Vec3f position;
+		float radialSpeed;
+	};
+	std::vector<TrackedObject> trackedObjects;
+	std::vector<int> objectIndexLookup(detectionsCount, -1);
```
Comment (prybicki) [inline src/graph/RadarTrackObjectsNode.cpp:49]: Leftover

File: src/graph/RadarTrackObjectsNode.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+	std::list<std::list<int>> objectIndices;
+	for (auto i = 0; i < detectionsCount; ++i) {
+		objectIndices.emplace_back(1, i);
+	}
```
Comment (prybicki) [inline src/graph/RadarTrackObjectsNode.cpp:54]: Please leave a little comment about meaning of those lists.

File: include/rgl/api/core.h [hunk 4]
```diff
@@ -828,6 +828,24 @@ RGL_API rgl_status_t rgl_node_points_radar_postprocess(rgl_node_t* node, const r
...
+ * Note that for correct calculation and publishing some of object characteristics (e.g. velocity) user has to call rgl_scene_set_time for current scene.
+ * Graph input: point cloud, containing RGL_FIELD_DISTANCE_F32, RGL_FIELD_AZIMUTH_F32, RGL_FIELD_ELEVATION_F32 and RGL_FIELD_RADIAL_SPEED_F32
+ * Graph output: point cloud
+ * @param node If (*node) == nullptr, a new Node will be created. Otherwise, (*node) will be modified.
+ * @param object_distance_threshold The maximum distance between any radar detection and other detection closest to it within the same tracked object (in meters).
+ * @param object_azimuth_threshold The maximum azimuth difference between any radar detection and other detection closest to it within the same tracked object (in radians).
+ * @param object_elevation_threshold The maximum elevation difference between any radar detection and other detection closest to it within the same tracked object (in radians).
+ * @param object_radial_speed_threshold The maximum radial speed difference between any radar detection and other detection closest to it within the same tracked object (in meters per second).
```
[the remark this answers is not in the document]
Comment (prybicki) [reply include/rgl/api/core.h:843]: Alternative: `The maximum distance between a radar detection and other closest detection classified as the same tracked object (in meters).`

File: src/graph/NodesCore.hpp [hunk 5]
```diff
@@ -588,6 +589,95 @@ struct RadarPostprocessPointsNode : IPointsNodeSingleInput
...
+{
+	using Ptr = std::shared_ptr<RadarTrackObjectsNode>;
+
+	enum class ObjectStatus : uint8_t
+	{
+		Measured = 0,
+		New = 1,
+		Predicted = 2,
+		Invalid = 255
+	};
```
[the remark this answers is not in the document]
Comment (PawelLiberadzki) [reply src/graph/NodesCore.hpp:602]: The reason here is simple - Measured == 0 matches radar documentation on object status, and uint8_t is the exact type of this data in raw frame. Changing to Invalid == 0 would result in making additional operation, when we will be filling the raw frame. Similar case is with enum type here.

File: src/math/RuningStats.hpp [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+// Integral types here generally make no sense in terms of mean and std dev.
+// I would force users to use floats, to eliminate possible errors.
+template<typename StatType, typename = std::enable_if_t<!std::is_integral_v<StatType>>>
```
[the remark this answers is not in the document]
Comment (PawelLiberadzki) [reply src/math/RuningStats.hpp:19]: This, rather just an overkill to make it 100% precise.

**Review by prybicki — APPROVED** [review]
:medal_military:

**Pull request merged**